fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb

```diff
@@ -103,12 +103,38 @@ def self.find_gsp_path(params)
       end
 
       def self.find_binary_path(params)
-        params[:binary_path] ||= (Dir["/Applications/Fabric.app/**/upload-symbols"] + Dir["./Pods/Fabric/upload-symbols"] + Dir["./scripts/upload-symbols"] + Dir["./Pods/FirebaseCrashlytics/upload-symbols"]).last
-        UI.user_error!("Failed to find Fabric's upload_symbols binary at /Applications/Fabric.app/**/upload-symbols or ./Pods/**/upload-symbols. Please specify the location of the binary explicitly by using the binary_path option") unless params[:binary_path]
+        params[:binary_path] ||= (Dir["/Applications/Fabric.app/**/upload-symbols"] + Dir["./Pods/Fabric/upload-symbols"] + Dir["./scripts/upload-symbols"] + Dir["./Pods/FirebaseCrashlytics/upload-symbols"] + Dir[spm_uploadsymbols_path]).last
+        UI.user_error!("Failed to find Fabric's upload_symbols binary at /Applications/Fabric.app/**/upload-symbols or ./Pods/**/upload-symbols or in the SPM SourcePackages folder. Please specify the location of the binary explicitly by using the binary_path option") unless params[:binary_path]
 
         params[:binary_path] = File.expand_path(params[:binary_path])
       end
 
+      def self.spm_uploadsymbols_path
+        config = {}
+        FastlaneCore::Project.detect_projects(config)
+        project = FastlaneCore::Project.new(config)
+        # This is the build path excluding the random string after the hyphen
+        build_base = "#{derived_data_path}/#{project.project_name}"
+        "#{build_base}-*/**/upload-symbols"
+      end
+
+      def self.derived_data_path
+        path = xcode_preferences ? xcode_preferences['IDECustomDerivedDataLocation'] : nil
+        path ||= "~/Library/Developer/Xcode/DerivedData"
+        # Make it absolute so it works with Dir later
+        File.expand_path(path)
+      end
+
+      def self.xcode_preferences
+        require 'fastlane_core/core_ext/cfpropertylist'
+        file = File.expand_path("~/Library/Preferences/com.apple.dt.Xcode.plist")
+        if File.exist?(file)
+          plist = CFPropertyList::List.new(file: file).value
+          return CFPropertyList.native_types(plist) unless plist.nil?
+        end
+        return nil
```

Comment:
```suggestion        
        plist = CFPropertyList::List.new(file: file).value if File.exist?(file)
        CFPropertyList.native_types(plist) unless plist.nil?
```

This would be simpler✂️